Add flipplatforms

Jontpan opened the pull request:
This commit adds a new platform type to the game, the flipplatform.
Jumping on it rotates the camera 180 degrees. Since PlatformSpawner was a child of the camera,
it would cause the platforms to be spawned below the player. Fixed this by
having the GameController be the same position as the camera but not a child,
created an empty platformspawner gameobject as a child on gamecontroller.

Rules for flipplatforms:

- It will only flip the first time the player collides with it.
- If spawned, there will always be another one to flip you back no more than 4 platforms after. It could be less though.

Since I for the first time made changes to the editor I pushed all changed files. Be sure to test it out.

Jontpan committed:
Add flipplatforms

This commit adds a new platform type to the game, the flipplatform.
Jumping on it rotates the camera 180 degrees. Since PlatformSpawner was a child of the camera,
it would cause the platforms to be spawned below the player. Fixed this by
having the GameController be the same position as the camera but not a child,
created an empty platformspawner gameobject as a child on gamecontroller.

Rules for flipplatforms:

- It will only flip the first time the player collides with it.
- If spawned, there will always be another one to flip you back no more than 4 platforms after. It could be less though.

AxlLind commented on 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs:
```diff
@@ -56,15 +56,70 @@ public class PlatformSpawnerScript : MonoBehaviour {
     GameObject PlatformScheme(int percentNormal, int percentTrampoline, int percentMoving)
```
Could these two methods perhaps be condensed to one? Or maybe this one makes a call like PlatformScheme(a,b,c, 0) to the other function.

Jontpan replied:
fixed in latest commit

AxlLind reviewed (commented):
The implementation looks good! 

Two things I think should be changed though:

* When the camera flips the x-movement doesn't which makes it very unintuitive/difficult. This makes it so that pressing right arrow makes you go left on the screen. The x-movement value should basically be multiplied by -1 while the screen is flipped.

* What I mentioned in the code comment below.

Jontpan commented:
Isn't it more intuitive as it is?

AxlLind commented:
I don't know, you might be right. With how it is now if you're going left before switching you continue going the same way but then after you've switch the movement is very difficult.

But honestly after testing it for a while I think it might just be too difficult over all.

Jontpan commented:
Too difficult? Shmoo Smhifficult! And next commit is gonna be smooooooth.

Jontpan committed:
Make cameraflipping flippin' smooth

Now makes the flipscripts call a coroutine in gamecontrollerscript that
rotates the camera in a lerp.

Jontpan commented:
Also the first flips are scheduled at y = 1000, should be fine with a challenge for those peeps.

AxlLind commented:
Oh wow, that flip animation looks great. I think what might be nice is stopping the player movement during the animation. That way you have just a split second to react and it might make it a bit more doable

Jontpan commented:
That's a good idea! Should we pause the rest of the game aswell?

AxlLind commented:
Yeah, could maybe be implemented by just calling the PauseUnPauseGame() function before and after the for loop in the co-routine.

Jontpan committed:
Add Pause function in Flip Animation

AxlLind committed:
Merge pull request #72 from AxlLind/flip-with-pause

Merge the two flip-platform branches

AxlLind commented:
I really like that actually now with the pause functionality. It gives you just a second to react. Will merge this.
